account_asset_management/models/account_asset.py

```diff
@@ -520,7 +520,12 @@ def name_get(self):
 
     def validate(self):
         for asset in self:
-            if asset.company_currency_id.is_zero(asset.value_residual):
+            if (
+                asset.company_currency_id.compare_amounts(
+                    asset.value_residual, asset.salvage_value
```

Comment:
But value_residual is purchase_value - salvage_value - value_depreciated 
https://github.com/OCA/account-financial-tools/blob/14.0/account_asset_management/models/account_asset.py#L323

Seems that you have an issue with your setup

Reply:
I think the asset should close when all depreciation journal entries are posted. In the case linear up to salvage method, when posting all depreciation journal entries, a residual value equal to salvage value which maybe not equal 0 so I think the asset close when residual value = salvage value will cover all cases. What do you think?

Reply:
I think value_residual = purchase_value - value_depreciated.